src/main/java/duke/sonautil/Storage.java

```diff
@@ -0,0 +1,226 @@
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+/**
+ * Works on loading, changing and maintaining the contents in the hard disk file
+ */
+public class Storage {
+    private final String path;
+
+    public Storage(String path) {
+        this.path = path;
+    }
+
+    /**
+     * Converts the command into hard disk file for record.
+     *
+     * @param command details of command retrieved from Parser
+     * @throws IOException if writeToFile/changeMarking fails
+     */
+    public void executeCommand(String[] command) throws IOException {
+
+        String keyword = command[0];
+        switch (keyword) {
+        case "todo":
+            writeToFile("T | 0 | " + command[1]);
+            break;
+        case "event":
+            writeToFile("E | 0 | " + command[1] + " | " + command[2]);
+            break;
+        case "deadline":
+            writeToFile("D | 0 | " + command[1] + " | " + command[2]);
+            break;
+        case "delete":
+            deleteFromFile(Integer.parseInt(command[1]));
+            break;
+        case "mark":
+            changeMarking("0", "1", Integer.parseInt(command[1]));
+            break;
+        case "unmark":
+            changeMarking("1", "0", Integer.parseInt(command[1]));
+            break;
+        default:
+        }
+    }
+
+    /**
+     * deletes the Duke.task from local file by creating a temp file
+     * and copy every Duke.task except the one to be deleted
+     *
+     * @param taskToDelete the index of Duke.task to delete
+     */
+    public void deleteFromFile(int taskToDelete) {
+        String oldFileName = "src/main/data/duke.txt";
+        String tempFileName = "src/main/data/tempDuke.txt";
+        File tempFile = new File(tempFileName);
+
+        int indexAccessed = 0;
+
+        try (BufferedReader br = new BufferedReader(new FileReader(oldFileName));
+             BufferedWriter bw = new BufferedWriter(new FileWriter(tempFile))) {
+
+            String line;
+            boolean first = true;
+            while ((line = br.readLine()) != null) {
+                if (indexAccessed != taskToDelete) {
+                    if (first) {
+                        bw.write(line);
+                        first = false;
+                    } else {
+                        bw.write("\n" + line);
+                    }
+                }
+                indexAccessed++;
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        File oldFile = new File(oldFileName);
+        oldFile.delete();
+
+        File newFile = new File(tempFileName);
+        newFile.renameTo(oldFile);
+    }
```

Comment:
I like your implementation of deleting tasks from the file, rather than rewriting the entire file. 👍